cli/src/cast.rs

```diff
@@ -134,6 +146,18 @@ async fn main() -> eyre::Result<()> {
                 cast_send(provider, from, to, sig, args, eth.cast_async).await?;
             }
         }
+        Subcommands::abiDecode { abi, hex_calldata, hex_output } => {
+            let parsed_abi = BaseContract::from(parse_abi(&[&abi]).unwrap());
```

Comment:
This function should be much simpler, something along the following lines:

```suggestion
let func = foundry_utils::IntoFunction::into(abi);
let data = hex::decode(calldata)
let decoded = func.decode_output(data)?;
println!("{}", decoded);
```

[`decode_output`](https://docs.rs/ethabi/latest/ethabi/struct.Function.html#method.decode_output) will return `Vec<Tokens>` based on the return types of the function